exercises/run-length-encoding/example.go

```diff
@@ -0,0 +1,55 @@
+package run_length_encoding
+
+import (
+	"fmt"
+	"strconv"
+)
+
+func RunLengthEncode(s string) string {
+	count := 1
+	output := ""
+	for i, c := range s {
+		if i != 0 {
+			if rune(s[i-1]) == c {
+				count++
+			} else {
+				getEncodedOutput(&count, i, s, &output)
+			}
+		}
+	}
+	if len(s) != 0 {
+		getEncodedOutput(&count, len(s), s, &output)
+	}
+	return output
+}
+
+func RunLengthDecode(s string) string {
+	count := 1
+	stringCount := ""
+	output := ""
+	for _, c := range s {
+		if _, err := strconv.Atoi(string(c)); err == nil {
+			stringCount += string(c)
+		} else {
+			if stringCount != "" {
+				count, _ = strconv.Atoi(stringCount)
+			}
+			for j := 0; j < count; j++ {
+				output += string(c)
+			}
+			count = 1
+			stringCount = ""
+		}
+	}
+	return output
+}
+
+func getEncodedOutput(count *int, i int, s string, output *string) {
```

Comment:
As noted [here](https://github.com/exercism/configlet/pull/83#discussion_r142034240), I'd change this function name. Maybe `encode`?

Reply:
I think this function would work better if it returned the updated variables, rather than altering pointers;

```
func encode(count int, i int, s string, output string) (int, string) {
	if count > 1 {
		output += fmt.Sprintf("%d%c", count, s[i-1])
		return 1, output
	}
	output += fmt.Sprintf("%c", s[i-1])
	return 1, output
}
```

you could then use it like so:

``
count, output = encode(count, i, s, output)
```

It's more explicit about what it's doing, and simplifies the function itself.